docs/docs/reference/extend/factored-instances.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: doc-page
+title: "Factored Instance Declarations"
+---
+
+Sometimes we want to re-use parts of one instance declaration in another. As an example,
+let's flesh out the `Functor` - `Monad` example, adding a typeclass trait `Applicative`
+for applicative functors in-between.
+
+```scala
+trait Functor[A] extends TypeClass {
+  def map[B](f: A => B): This[B]
+}
+
+trait Applicative[A] extends Functor[A] {
+  def map2[B, C](that: This[B])(f: (A, B) => C): This[C]
+  common def pure[B](x: B): This[B]
+
+  def mapWith[B](f: This[A => B]): This[B] = map2(f)((x, f) => f(x))
+  def map[B](f: A => B): This[B] = mapWith(pure(f))
+}
+
+trait Monad[A] extends Applicative[A] {
+  def flatMap[B](f: A => This[B]): This[B]
+}
+```
+`Applicative` can be extended in other ways as well. Here is `Traverse`, another common sub-trait of `Applicative`:
```

Comment:
The statement that `Traverse` is a common sub-trait of `Applicative` doesn't seem right to me. In all of the contexts that I'm familiar with (Haskell, Cats, Scalaz, Purescript), `Traverse`/`Traversable` extend `Functor` and `Foldable` but not `Applicative`.

Reply:
Right. I have to revise the example.